### PR TITLE
Skip promoting new stable if current stable is not `machine=True`

### PR DESCRIPTION
**Pull request by humitos**
When the "stable" is managed by the user (because they have a
`origin/stable` branch, for example) we do not need to calculate which
would be the version to be promoted as stable.

So, we return immediately (without doing any computation) if the
current stable is not managed by Read the Docs.

### CODE REVIEW

File: readthedocs/projects/models.py [hunk 1]
```diff
@@ -1022,15 +1022,21 @@ def update_stable_version(self):
...
+
         versions = self.versions(manager=INTERNAL).all()
         new_stable = determine_stable_version(versions)
         if new_stable:
-            current_stable = self.get_stable_version()
             if current_stable:
                 identifier_updated = (
                     new_stable.identifier != current_stable.identifier
                 )
-                if identifier_updated and current_stable.machine:
```
Comment (ericholscher) [inline readthedocs/projects/models.py]: Won't this break users who are setting `stable` manually?

Comment (ericholscher) [reply readthedocs/projects/models.py]: Oh, I see how it works now. Ignore me :)

**Comment by humitos** [comment]
This is valid. I just need to write some tests for it.

**Comment by ericholscher** [comment]
@humitos would be useful to get the tests fixed up here, so we can ship it 👍

**Pull request merged**